File: Java/ObjectPrograms/DictionaryBST/DictionaryTest.java
/* DictionaryTest.java
 * Sean Gordon
 * skgordon #1405355
 * CMPS 12B pa3
 * 4/26/15
 * Test file for Dictionary ADT
 */

class DictionaryTest{
    public static void main(String[] args){
        Dictionary D = new Dictionary();
        String v;

        System.out.println("==========Section 1==========");
        D.insert("1", "a");
        D.insert("2", "b");
        v = D.lookup("1");
        System.out.println("key = 1 "+(v==null?"not found":("value = "+v)));
        v = D.lookup("2");
        System.out.println("key = 2 "+(v==null?"not found":("value = "+v)));

        System.out.println("==========Section 2==========");
        System.out.println(D.isEmpty());
        System.out.println(D.size());
        D.insert("Beer", "blue moon");
        D.insert("Weed", "grape ape");
        D.insert("Dank", "bud");
        System.out.println(D.size());

        System.out.println("==========Section 3==========");
        D.delete("2");
        System.out.println(D.size());
        System.out.println();
        System.out.println(D);
        System.out.println("Lookup for key 'Weed': "+D.lookup("Weed"));
        System.out.println("Lookup for key '1': "+D.lookup("1"));

        System.out.println("==========Section 4==========");
        System.out.println(D.size());
        D.delete("1");
        System.out.println(D.size());

        System.out.println("==========Section 5==========");
        D.makeEmpty();
        System.out.println(D.isEmpty());
        System.out.println(D.size());
        D.insert("Country", "Capital");
        D.insert("USA", "DC");
        D.insert("France", "Paris");
        System.out.println(D.size());
        System.out.println();
        System.out.println(D);

        // These will produce errors
        System.out.println("==========Section 6==========");
        // D.insert("Country", "Spain");
        // D.delete("Good");
    }
} 

File: C/Dictionary/DictionaryTest.c
/* DictionaryTest.c
 * Sean Gordon
 * skgordon #1405355
 * CMPS 12B pa5
 * 5/30/15
 * Test file for the Hash Table Dictionary ADT
 */

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
    Dictionary A = newDictionary();

    printf("==========Section 1==========\n");
    insert(A, "one", "une");
    insert(A, "two", "deux");
    insert(A, "three", "trois");
    insert(A, "four", "quatre");
    insert(A, "five", "cinq");
    printf("Size of A: %d\n", size(A));
    printf("Is A empty?: %s\n", (isEmpty(A)?"true":"false"));
    printf("Lookup for 'four': %s\n", lookup(A, "four"));
    printDictionary(stdout, A);

    printf("==========Section 2==========\n");
    delete(A, "two");
    delete(A, "four");
    printf("Size of A: %d\n", size(A));
    printf("Is A empty?: %s\n", (isEmpty(A)?"true":"false"));
    makeEmpty(A);
    printf("Size of A: %d\n", size(A));
    printf("Is A empty?: %s\n", (isEmpty(A)?"true":"false"));
    freeDictionary(&A);

    printf("==========Section 3==========\n");
    Dictionary B = newDictionary();
    insert(B, "France", "Paris");
    insert(B, "Spain", "Madrid");
    insert(B, "Germany", "Berlin");
    insert(B, "Britain", "London");
    printf("Size of B: %d\n", size(B));
    printf("Is B empty?: %s\n", (isEmpty(B)?"true":"false"));
    delete(B, "France");
    printDictionary(stdout, B);

    printf("==========Section 4==========\n");
    printf("This section will cause errors if uncommented\n");
    //printf("Lookup for 'one': %s\n", lookup(A, "one"));
    //insert(B, "Germany", "Munich");
    //delete(B, "five");
    //delete(B, "France");

    printf("==========Section 5==========\n");
    freeDictionary(&B);

    return(EXIT_SUCCESS);
}
